src/RA_Core.cpp

```diff
@@ -463,6 +460,14 @@ API int CCONV _RA_OnLoadNewRom(const BYTE* pROM, unsigned int nROMSize)
     return 0;
 }
 
+API void CCONV _RA_OnReset()
+{
+    g_LeaderboardManager.Reset();
+    g_PopupWindows.LeaderboardPopups().Reset();
+
+    g_nProcessTimer = 0;
+}
```

Comment:
on reset, clear any popups, disable active leaderboard challenges, and reset the achievement processing delay